src/AnimalClub/Skellygen/Mesh.hs

```diff
@@ -47,28 +45,34 @@ type Face = (Int32,Int32,Int32)
 
 -- indices index all the other data in this representation
 data PotatoMesh a = PotatoMesh {
-  positions   :: [V3 a]
-  , normals   :: [V3 a]
-  , texCoords :: [V2 a]
-  , indices   :: [Face]
-}
+  positions   :: V.Vector (V3 a)
+  , normals   :: V.Vector (V3 a)
+  , texCoords :: V.Vector (V2 a)
+  , indices   :: V.Vector Face
+} deriving (Generic, NFData)
+
+emptyPotatoMesh :: (AnimalFloat a) => PotatoMesh a
+emptyPotatoMesh = PotatoMesh V.empty V.empty V.empty V.empty
+
+
+-- this can probably be improved even more to take advantage of stream fusion
+concatPotatoMesh :: (AnimalFloat a) => [PotatoMesh a] -> PotatoMesh a
+concatPotatoMesh pms = r where
+  foldfn (PotatoMesh p n tc i) (pl, nl, tcl, il, offset) =
+    (pl++[p], nl++[n], tcl++[tc], il++[offseti], newOffset) where
+      offseti = V.map (map3Tuple (+ fromIntegral offset)) i
+      newOffset = offset + V.length p
+  (plf,nlf,tclf,ilf,_) = foldr foldfn ([],[],[],[],0) pms
```

Comment:
`V.concat` is of type `[Vector a] -> Vector a`
and I have `[PotatoMesh a]` so this step converts it into a tuple of `[Vector (V3 a)]`s and also offsets the indices

Reply:
I thought `foldl'` was better to use than `foldr` unless you needed the laziness.

Reply:
Good point. I should get use to using foldl' more. I'll change it in the future because I already made a bunch of commits on top of this PR 🙀